classes/form/enrolkey_signup_form.php

```diff
@@ -78,49 +78,66 @@ public function definition() {
      *         or an empty array if everything is OK (true allowed for backwards compatibility too).
      */
     public function validation($data, $files) {
-        global $DB;
         $errors = parent::validation($data, $files);
 
-        $enrolplugin = enrol_get_plugin('self');
+        $signuptoken = $data['signup_token'];
 
-        $token = $data['signup_token'];
+        if ($signuptoken !== '') {
+            // For any case where the token is populated, perform a lookup.
+            $tokenisvalid = $this->check_database_for_enrolkey($signuptoken);
 
-        if (!empty($token)) {
-            $selfenrolinstance = false;
+            if ($tokenisvalid === false) {
```

Comment:
could mildy simplify to !$tokenisvaldid

Reply:
I feel that this check is more verbose, the reason why this patch exists was to get away from !empty() evaluating the character '0' as true in the earlier statement. I'd like to be able to read what is the exact condition.